csrc/scheduler/registry_utils.cpp

```diff
@@ -449,53 +449,36 @@ bool reductionInterferingView(
 
   NVF_ERROR(dims.empty(), "Error processing ", dims, " in registry.cpp.");
 
-  // Make sure groups are disjoint based on view
-
-  auto disjoint_rfactor_sets = scheduler_utils::disjointLogicalSets(fusion);
-  auto disjoint_set_information = scheduler_utils::getDisjointLogicalSetsOf(
-      fusion, reduction_reference, disjoint_rfactor_sets);
-
-  // Convert id's in groups to disjoint_set_ids of disjoint_set_information
-  std::vector<std::vector<int64_t>> disjoint_groups;
-
-  for (const auto& group : groups) {
-    std::vector<int64_t> disjoint_id_sets;
-    for (auto id : group) {
-      auto find_it = std::find(
-          reduction_reference->getLogicalDomain().begin(),
-          reduction_reference->getLogicalDomain().end(),
-          id);
-      NVF_ERROR(
-          find_it != reduction_reference->getLogicalDomain().end(),
-          "Issue with view analysis on reduction like schedule, with reference: ",
-          reduction_reference->toString());
-      auto logical_pos = std::distance(
-          reduction_reference->getLogicalDomain().begin(), find_it);
-      NVF_ERROR(
-          logical_pos <
-              (int64_t)disjoint_set_information.disjoint_set_ids.size(),
-          "Error computing disjoint group on the logical domain of ",
-          reduction_reference->toString());
-      disjoint_id_sets.push_back(
-          disjoint_set_information.disjoint_set_ids[logical_pos]);
+  // Check if any view op could merge IDs in different groups
+  IdModel id_model(fusion, false, false, false);
+  id_model.buildExactGraph();
+  const auto& id_use_map = id_model.idUses();
+  const ValGraph& exact_graph = id_model.idGraph(IdMappingMode::EXACT);
+
+  // ValGroup to group index
+  std::unordered_map<ValGroup, int> vg_to_group_idx;
+  for (auto group_id : c10::irange(groups.size())) {
+    for (auto id : groups.at(group_id)) {
+      auto vg = exact_graph.toGroup(id);
+      vg_to_group_idx[vg] = (int)group_id;
     }
-    disjoint_groups.push_back(disjoint_id_sets);
   }
 
-  // Make sure there's no intersection between the groups, otherwise view
-  // will interfere with the schedule. TODO: Make this better complexity,
-  // since it should be relatively small int vectors of a small total nDims,
-  // not too worried about it now.
-
-  for (auto first_dim_i : c10::irange(disjoint_groups.size())) {
-    for (auto second_dim_i = first_dim_i + 1;
-         second_dim_i < disjoint_groups.size();
-         ++second_dim_i) {
-      auto first_group = disjoint_groups[first_dim_i];
-      auto second_group = disjoint_groups[second_dim_i];
-      for (auto first_disjoint_id : first_group) {
-        for (auto second_disjoint_id : second_group) {
-          if (first_disjoint_id == second_disjoint_id) {
+  // Loop through all view ops and check if they merge IDs in different groups
+  // Map through: ID --> ValGroup --> Group index
```

Comment:
I think I understand it as that, each group would be a *merged* id for scheduling to handle, so merge on IDs across group would break scheduler later.

Maybe it's just me. I'm wondering if we should put an explicit comment on that here. Since it's easy to mistakenly associate `group_idx` with ValGroup and get very confused (like I was earlier)?

Reply:
Good point. renamed it to `vect_of_coalesced_ids` and added comments.
```
  // Each vector of IDs is a group of coalesced_ids, they will be merged into
  // one ID during the reduction schedule. If the view op involves merging IDs
  // in different group of coalesced_ids, it would break the reduction
  // scheduler.
  std::vector<std::vector<IterDomain*>> vect_of_coalesced_ids;
```